packages/core/ios/EmbraceManager.swift

```diff
@@ -21,83 +22,94 @@ private let EVENT_ATTRIBUTES_KEY = "attributes"
-        } catch let error {
-          reject("SET_JS_BUNDLE_PATH_ERROR", "Error setting JavaScript bundle path", error)
+    @objc(setJavaScriptBundlePath:resolver:rejecter:)
+    func setJavaScriptBundlePath(_ path: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+        DispatchQueue.global(qos: .background).async {
+            do {
+                let bundleID = try computeBundleID(path: path)
+                try Embrace.client?.metadata.addResource(key: REACT_NATIVE_BUNDLE_ID_RESOURCE_KEY, value: bundleID.id, lifespan: .process)
+                resolve(true)
+            } catch let error {
+                reject("SET_JS_BUNDLE_PATH_ERROR", "Error setting JavaScript bundle path", error)
+            }
         }
     }
-  }
+
     @objc
     func getDefaultJavaScriptBundlePath(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        if  let filePath = Bundle.main.path(forResource: "main", ofType: "jsbundle"){
+        if  let filePath = Bundle.main.path(forResource: "main", ofType: "jsbundle") {
             resolve(filePath)
         } else {
             reject("error", "Unable to retrieve JS bundle path", nil)
         }
     }
+
     @objc
     func isStarted(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
         if let embraceStarted = Embrace.client?.started {
-        resolve(embraceStarted)
+            resolve(embraceStarted)
         } else {
-        resolve(false)
+            resolve(false)
         }
     }
 
     @objc(startNativeEmbraceSDK:resolver:rejecter:)
     func startNativeEmbraceSDK(configDict: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
-        let config = SDKConfig(from: configDict)
+        config = SDKConfig(from: configDict)
+        
         DispatchQueue.main.async {
             do {
                 var embraceOptions: Embrace.Options {
                     var crashReporter: CrashReporter?
-                    if config.disableCrashReporter {
+                    if self.config.disableCrashReporter {
                         crashReporter = nil
                     } else {
                         crashReporter = EmbraceCrashReporter()
                     }
 
                     let servicesBuilder = CaptureServiceBuilder().addDefaults()
-                    if config.disableAutomaticViewCapture {
+                    if self.config.disableAutomaticViewCapture {
                             servicesBuilder.remove(ofType: ViewCaptureService.self)
                     }
 
+                    if self.config.disableNetworkSpanForwarding {
+                        servicesBuilder.add(.urlSession(options:
+                                                            URLSessionCaptureService.Options(injectTracingHeader: false, requestsDataSource: nil)))
```

Comment:
just want to confirm my understanding here, so is it the case that by default with `CaptureServiceBuilder().addDefaults()` we get a  `URLSessionCaptureService` that has injecting tracing headers on? And then this replaces that service rather than adding a 2nd one?

Reply:
because of this comment I've dug into this scenario. As per what I read in the ios sdk repo is [here](https://github.com/embrace-io/embrace-apple-sdk/blob/7ae30ac688c6d3cf8c19e923833812ce10457326/Sources/EmbraceIO/Capture/CaptureServiceBuilder.swift#L40) I think yes, it would add a 2nd service so I've refactored it a little bit to be sure the service is not added twice. I'll perform some manual tests and see if I can introduce tweaks that tests this situation during unit tests if possible